dnsimple/authentication.go

```diff
@@ -1,9 +1,24 @@
 package dnsimple
 
 import (
+	"context"
 	"net/http"
+
+	"golang.org/x/oauth2"
 )
 
+// BasicAuthClient returns a client that authenticates via HTTP Basic Auth with given username and password.
+func BasicAuthClient(_ context.Context, username, password string) *http.Client {
+	tp := BasicAuthTransport{Username: username, Password: password}
+	return tp.Client()
+}
+
+// BasicAuthClient returns a client that authenticates with a static OAuth token.
+func StaticTokenClient(ctx context.Context, token string) *http.Client {
```

Comment:
Given this is an HTTP client, consider including that in the name of the function. `StaticTokenHTTPClient`